conda/cli/install.py

```diff
@@ -64,7 +67,57 @@
 stderrlog = getLogger("conda.stderr")
 
 
-def check_prefix(prefix, json=False):
+def validate_prefix_exists(prefix: str | Path) -> Path:
+    """
+    Validate that we are receiving at least one valid value for --name or --prefix.
+    """
+    prefix = Path(prefix)
+    if not prefix.exists():
+        raise CondaEnvException("The environment you have specified does not exist.")
+
+    return prefix
+
+
+def check_protected_dirs(prefix: str | Path, json: bool = False) -> None:
+    """Ensure that the new prefix does not contain protected directories."""
+    parent_dir = Path(prefix).parent.resolve()
+
+    if exists(parent_dir):
+        conda_meta_dir = os.path.join(parent_dir, "conda-meta")
+        if os.path.isdir(conda_meta_dir):
+            history_file = os.path.join(conda_meta_dir, "history")
+            if os.path.isfile(history_file):
+                raise CondaEnvException(
+                    f"The specified prefix '{prefix}' "
+                    "contains a protected directory, 'conda-meta', and a 'history' file within it. "
+                    "Creating an environment in this location will potentially risk the deletion "
+                    "of other conda environments. Aborting.",
+                    json,
+                )
```

Comment:
Should we use pathlib for this as well?

```suggestion
def check_protected_dirs(prefix: str | Path, json: bool = False) -> None:
    """Ensure that the new prefix does not contain protected directories."""
    parent_dir = Path(prefix).parent.resolve()

    if parent_dir.exists():
        conda_meta_dir = parent_dir / "conda-meta"
        if conda_meta_dir.is_dir():
            history_file = conda_meta_dir / "history"
            if history_file.is_file():
                raise CondaEnvException(
                    f"The specified prefix '{prefix}' "
                    "contains a protected directory, 'conda-meta', and a 'history' file within it. "
                    "Creating an environment in this location will potentially risk the deletion "
                    "of other conda environments. Aborting.",
                    json,
                )
```